code/src/components/Todolist.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import todos from "../reducers/todos";
+
+const TodoList = () => {
+  const items = useSelector((store) => store.todos.items);
+
+  const dispatch = useDispatch();
+
+  const onToggleTodo = (id) => {
+    dispatch(todos.actions.toggleTodo(id));
+  };
+
+  const onDeleteTodo = (id) => {
+    dispatch(todos.actions.deleteTodo(id));
+  };
+
+  const onCompleteTodo = (id) => {
+    dispatch(todos.actions.toggleTodo(id));
+  };
+
+  const onDoingTodo = (id) => {
```

Comment:
I'm not sure what it is but your toggles are affecting each other. When you check one of them "done" or "doing" the other one gets checked aswell and they toggle each other. So there is something off with the action in the reducer and how you are using it in the TodoList.. 

I belieive it could be cause your using the exact same funtion in your reducer and you are passing the same info to the function. The done and doing gets have the same id so they toggle each other out I guess..?